File: articles/stp-segmentation-targeting-and-positioning/r/run_all_stp_workflows.R
# Run all STP R workflows.

source(file.path("r", "stp_segment_positioning_report.R"))
source(file.path("r", "stp_dimension_summary.R"))
source(file.path("r", "stp_targeting_plots.R"))

cat("All STP R workflows completed.\n")

File: articles/stp-segmentation-targeting-and-positioning/python/run_all_stp_workflows.py
#!/usr/bin/env python3
"""Run all Python STP workflows in sequence."""

from __future__ import annotations

import subprocess
import sys
from pathlib import Path

HERE = Path(__file__).resolve().parent

scripts = [
    "stp_audience_positioning_audit.py",
    "segment_priority_model.py",
    "positioning_gap_checker.py",
    "ethical_targeting_review.py",
]

for script in scripts:
    print(f"Running {script}...")
    subprocess.run([sys.executable, str(HERE / script)], check=True)

print("All STP Python workflows completed.")
